base/intfuncs.jl

```diff
@@ -149,8 +155,8 @@ lcm(a::Real, b::Real, c::Real...) = lcm(a, lcm(b, c...))
 gcd(a::T, b::T) where T<:Real = throw(MethodError(gcd, (a,b)))
 lcm(a::T, b::T) where T<:Real = throw(MethodError(lcm, (a,b)))
 
-gcd(abc::AbstractArray{<:Real}) = reduce(gcd, abc; init=zero(eltype(abc)))
-lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc; init=one(eltype(abc)))
+gcd(abc::AbstractArray{<:Real}) = reduce(gcd, abc)
+lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc)
```

Comment:
```suggestion
lcm(abc::AbstractArray{<:Integer}) = reduce(lcm, abc, init=one(eltype(abc)))
lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc)
```
For integers, 1 is an identity of LCM (because all integers are a multiple of 1) and is also the correct answer for the empty case (see https://github.com/JuliaLang/julia/issues/55379#issuecomment-2429406611).